.github/workflows/build_and_functional_tests.yml

```diff
@@ -23,11 +23,36 @@ jobs:
     uses: LedgerHQ/ledger-app-workflows/.github/workflows/reusable_build.yml@v1
     with:
       upload_app_binaries_artifact: "ragger_elfs"
-      flags: "DEBUG=1 CAL_TEST_KEY=1 DOMAIN_NAME_TEST_KEY=1 SET_PLUGIN_TEST_KEY=1 NFT_TEST_KEY=1"
+      flags: "CAL_TEST_KEY=1 DOMAIN_NAME_TEST_KEY=1 SET_PLUGIN_TEST_KEY=1 NFT_TEST_KEY=1"
 
   ragger_tests:
     name: Run ragger tests using the reusable workflow
     needs: build_application
     uses: LedgerHQ/ledger-app-workflows/.github/workflows/reusable_ragger_tests.yml@v1
     with:
       download_app_binaries_artifact: "ragger_elfs"
+
+  build_clone_app:
+    name: Build Clone application using the reusable workflow
+    uses: LedgerHQ/ledger-app-workflows/.github/workflows/reusable_build.yml@v1
+    with:
+      upload_app_binaries_artifact: "clone_elfs"
+
+  build_main_app:
+    name: Build Main app using the reusable workflow
+    uses: LedgerHQ/ledger-app-workflows/.github/workflows/reusable_build.yml@v1
+    with:
+      flags: "CHAIN=thundercore"
+      upload_app_binaries_artifact: "main_elfs"
+
+  ragger_clone_tests:
+    name: Run ragger tests using the reusable workflow for lib context
+    needs:
+      - build_clone_app
+      - build_main_app
+    uses: LedgerHQ/ledger-app-workflows/.github/workflows/reusable_ragger_tests.yml@v1
+    with:
+      download_app_binaries_artifact: "clone_elfs"
+      additional_app_binaries_artifact: "main_elfs"
+      additional_app_binaries_artifact_dir: ./tests/.test_dependencies/main/build/
```

Comment:
I would prefer if this was moved to `./tests/ragger/.test_dependencies/clone/build/` since it is Ragger-specific.

Reply:
Done